cli/src/cli_util.rs

```diff
@@ -1302,6 +1316,38 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
             let failed_branches = git::export_refs(tx.mut_repo())?;
             print_failed_git_export(ui, &failed_branches)?;
         }
+
+        if let Some(new_commit) = &maybe_new_wc_commit {
+            if self
+                .check_repo_rewritable(tx.mut_repo(), [new_commit.id()])
+                .is_err()
+            {
+                let mut new_wc_commit: Option<Commit> = None;
+                tx.mut_repo()
+                    .view()
+                    .wc_commit_ids()
+                    .clone()
+                    .iter()
+                    .try_for_each::<_, Result<(), CheckOutCommitError>>(|(workspace_id, _)| {
+                        let new = tx.mut_repo().check_out(
+                            workspace_id.clone(),
+                            &self.settings,
+                            new_commit,
```

Comment:
This should not be `new_commit` for the other workspaces. Actually, that suggests that this code should be moved a bit earlier. Probably just before where we assign `old_repo` is a good place. Then `maybe_new_wc_commit` can become immutable again. You can also move the code into its own function if that helps with readability.

And yes, please do add a test with multiple workspaces becoming immutable :)